File: backend/FamilyShop.Domain/Entities/Product.cs
using FamilyShop.Domain.ValueObjects;

namespace FamilyShop.Domain.Entities;

public enum Gender
{
    Male,
    Female,
    Kids
}

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Money Price { get; set; }
    public Money? DiscountPrice { get; set; }
    public int Stock { get; set; }
    public int CategoryId { get; set; }
    public Category? Category { get; set; }
    public Gender Gender { get; set; }
    public List<string> Images { get; set; } = new();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool IsBestseller { get; set; }

    // Denormalized from Review rows and recomputed whenever a review is added/removed,
    // so catalog listing/sorting never has to aggregate Reviews per page.
    public decimal AverageRating { get; set; }
    public int ReviewCount { get; set; }

    public Money EffectivePrice => DiscountPrice ?? Price;
}

File: frontend/src/types/product.ts
export type Gender = 'female' | 'male' | 'kids';

export interface Category {
  id: string;
  name: string;
  slug: string;
  hasSizes: boolean;
}

export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  discountPrice?: number;
  stock: number;
  categoryId: string;
  gender: Gender;
  images: string[];
  sizes: string[];
  isNew?: boolean;
  isBestseller?: boolean;
  createdAt: string;
  averageRating: number;
  reviewCount: number;
}
